logger/src/main/java/com/orhanobut/logger/FileLogger.java

```diff
@@ -0,0 +1,362 @@
+        LEVELS.append(Log.ASSERT, "ASSERT");
+    }
+    //endregion
+
+    private final Handler handler;
+    private final StringBuilder stringBuilder = new StringBuilder(200);
+    private final SimpleDateFormat dateFormat;
+    private final Date date = new Date();
+    private final String separator;
+    private final int logLevel;
+
+    private FileLogger(int logLevel, int maxLines, String folder, String separator,
+                       String dateFormat, Locale locale, Looper looper) {
+        this.logLevel = logLevel;
+        this.separator = separator;
+        this.dateFormat = new SimpleDateFormat(dateFormat, locale);
+        this.handler = new WriteHandler(looper, folder, maxLines);
+    }
+
+    @Override
+    public void d(String tag, String message) {
+        log(Log.DEBUG, tag, message);
+    }
+
+    @Override
+    public void e(String tag, String message) {
+        log(Log.ERROR, tag, message);
+    }
+
+    @Override
+    public void w(String tag, String message) {
+        log(Log.WARN, tag, message);
+    }
+
+    @Override
+    public void i(String tag, String message) {
+        log(Log.INFO, tag, message);
+    }
+
+    @Override
+    public void v(String tag, String message) {
+        log(Log.VERBOSE, tag, message);
+    }
+
+    @Override
+    public void wtf(String tag, String message) {
+        log(Log.ERROR, tag, message);
+    }
+
+    //region message builder
+    private synchronized void log(int level, String tag, String message) {
+        if (logLevel <= 0 || level < logLevel) return;
+
+        if (MARKERS.contains(message)) {
+            if (BEGIN.equals(message)) {
+
+                stringBuilder.setLength(0);
+                date.setTime(System.currentTimeMillis());
+
+                // machine-readable date/time
+                stringBuilder.append(date.getTime());
+
+                // human-readable date/time
+                stringBuilder.append(separator);
+                stringBuilder.append(dateFormat.format(date));
+
+                // level
+                stringBuilder.append(separator);
+                stringBuilder.append(LEVELS.get(level));
+
+                // tag
+                stringBuilder.append(separator);
+                stringBuilder.append(tag);
+
+
+            } else if (END.equals(message)) {
+
+                stringBuilder.append(NEW_LINE);
+                handler.sendMessage(handler.obtainMessage(0, stringBuilder.toString()));
+
+            }
+            return;
+        }
+
+        if (message.contains(NEW_LINE)) {
+            message = message.replace(NEW_LINE, NEW_LINE_REPLACEMENT);
+        }
+
+        message = message.replace(HORZ_LINE, "").trim();
+
+        // messages
+        stringBuilder.append(separator);
+        stringBuilder.append(message);
+
+    }
+    //endregion
+
+    //region helper to generate file names
+    private static File getLogFile(String folderName, String fileName, int maxLines) {
```

Comment:
That's true, missed the handler thread part. Since it happens in a background thread, trying to optimize would not be that much important.